mlflow/recipes/steps/train.py

```diff
@@ -398,16 +441,22 @@ def my_warn(*args, **kwargs):
 
             target_data = raw_validation_df[self.target_col]
             prediction_result = model.predict(raw_validation_df.drop(self.target_col, axis=1))
-            if self.predict_proba:
-                prediction_result_probs = model.predict_proba(
-                    raw_validation_df.drop(self.target_col, axis=1)
+
+            use_predict_proba = isinstance(prediction_result, pd.DataFrame)
+            if use_predict_proba:
+                prediction_result_for_error = (
+                    prediction_result.drop(
+                        [f"{self.predict_prefix}label", f"{self.predict_prefix}score"], axis=1
```

Comment:
Can we check to make sure that these columns exist before attempting to drop / access them?